cosmos/dbt/graph.py

```diff
@@ -467,6 +468,13 @@ def load_via_dbt_ls_without_cache(self) -> None:
                 env[DBT_TARGET_PATH_ENVVAR] = str(self.target_dir)
 
                 if self.render_config.dbt_deps and has_non_empty_dependencies_file(self.project_path):
+                    if self.execution_config.execution_mode in (
+                        ExecutionMode.LOCAL,
+                        ExecutionMode.VIRTUALENV,
+                    ) and is_cache_lockfile_enabled(project_path):
```

Comment:
Execution config is related to the task execution; we should avoid it wherever possible in the rendering phase (when we parse dbt).

Since we're already in `load_via_dbt_ls_without_cache`, I believe it would be safe to do:
```suggestion
                    if is_cache_lockfile_enabled(project_path):
```